File: submissions/sm_026_rohit-kumar/week_03/day_4/session_1/mirrorImage.js
/*
 return mirror image of a single character
 return same char if mirror is not found
 */
function mirror_of(char) {
    caps  = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
    small = "abcdefghijklmnopqrstuvwxyz"
    
    var index = caps.indexOf(char)
    if (index >= 0) {
      return caps[25 - index]
    }
    else {
      index = small.indexOf(char)
      if(index >= 0){
        return small[25 - index]
      }
    }
    return char
}

// function to accumulate mirror character
function get_mirror_image(string) {
    new_string = ''
    for(var i = 0; i < string.length; i++) {
        new_string = new_string + mirror_of(string[i])
    }

    return new_string
}

// code execution starts here

var string = "PaPa"
var string = "PaPa"
var string = "AbKbA"

console.log(get_mirror_image(string))

/*
------------------------Test Cases-----------------------

Input --------------------------- Ouput

AyzB                              AyzB
PaPa                              KzKz
AbRaM                             ZyIzN
AbKbA                             ZyPyZ


*/

File: submissions/sm_103_apoorva/week_13/day_4/mirror_image.py
string = input("Enter the string: ")
# print(string)
out = ""
small = ['a','b','c','d','e','f','g','h','i','j','k','l','m','n','o','p','q','r','s','t','u','v','w','x','y','z']
caps = ['A','B','C','D','E','F','G','H','I','J','K','L','M','N','O','P','Q','R','S','T','U','V','W','X','Y','Z']
for i in string:
    if i >= "A" and i <= "Z":
        if i in caps:
            out += caps[25-caps.index(i)]
    else:
        if i in small:
            out += small[25-small.index(i)]
print(out)
